### PR TITLE
NMS-13339: Add FtpDetector docs

### PR DESCRIPTION
**Pull request by mfuhrmann**
### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [x] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-13339

### CODE REVIEW

File: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+= FTP Service Detector
+
+The FtpDetector is able to validate ftp connection dial-up processes.
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc:3]:
```suggestion
This detector lets you find FTP services for monitoring in your network.
```

File: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+The detector can test ftp server on multiple ports and specific login data.
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc:4]:
```suggestion
It finds FTP server on multiple ports and can use also login credentials for service detection.
```

File: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+|===
+
+== Configuration and Use
+
+.Parameters for the FTP Service Detector
+[options="header, %autowidth"]
+|===
+| Parameter            | Description   | Required | Default value
+| `port`               | Port to query | optional | `21`
+| `multilineIndicator` | ???           | optional | `-`
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc:20]: ???

File: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+|===
+
+== Example Configuration
+
+[source,xml]
+----
+<detector name="FTP" class="org.opennms.netmgt.provision.detector.simple.FtpDetector">
+  <parameter key="port" value="21"/>
+  <parameter key="multilineIndicator" value="-"/>
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc:29]: Maybe adding an example with credentials?

File: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc (hunk 2 above)
Comment (indigo423) [reply docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc:4]: We detect a FTP service using the command channel (21/tcp), right?

**Review by indigo423 — COMMENTED** [review]
Minor comments

**Review by Bonrob2 — COMMENTED** [review]
Aside from Ronny's comments, LGTM!

File: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc [hunk 5]
```diff
@@ -1,6 +1,7 @@
 = FTP Service Detector
 
-Use the FTP Service Detector to detect FTP services.
+The FtpDetector is able to validate ftp connection dial-up processes.
+The detector can test ftp server on multiple ports and specific login data.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc:4]:
```suggestion
It finds the FTP server on multiple ports and can also use login credentials for service detection.
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]

File: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc (hunk 2 above)
Comment (mfuhrmann) [reply docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc:4]: Yes. https://github.com/OpenNMS/opennms/blob/7e9f5759badaf5636ae487a72b605a2a89e3e13e/opennms-provision/opennms-detector-lineoriented/src/main/java/org/opennms/netmgt/provision/detector/simple/FtpDetector.java#L46

File: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc (hunk 3 above)
Comment (mfuhrmann) [reply docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc:20]: Feel free to add a useful description :-) I am not sure what exactly this is doing. The [monitor](https://github.com/OpenNMS/opennms/blob/develop/docs/modules/operation/pages/service-assurance/monitors/FtpMonitor.adoc) does not have this parameter.

File: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc (hunk 4 above)
Comment (mfuhrmann) [reply docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc:29]: I thought this is not possible due to the missing parameters here:
image

**Review by Bonrob2 — APPROVED** [review]

**Review by indigo423 — COMMENTED** [review]
If we have figured out how to deal with the "unknown" multilineIndicator we can go ahead and merge.

File: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc (hunk 3 above)
Comment (mfuhrmann) [reply docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc:20]:
```suggestion
```

**Review by indigo423 — APPROVED** [review]
If possible, would be nice to get rid of the multilineIndicator in the example and then 🚢 :D

File: docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc (hunk 4 above)
Comment (mfuhrmann) [reply docs/modules/operation/pages/provisioning/detectors/FtpDetector.adoc:28]:
```suggestion
```

**Review by Bonrob2 — APPROVED** [review]
LGTM!

**Pull request merged**